From d19ac283baf78e691d2e654a4284ddae87a1a714 Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Wed, 23 May 2018 16:06:31 +0300
Subject: [PATCH 3837/5242] MLK-18181-1 ARM: imx: Clear OF_POPULATED in
 imx_gpcv2_init

commit  80ae2bc5a9fd0cd4b1efca4a7e50f5d5dac6f7e2 from
https://source.codeaurora.org/external/imx/linux-imx.git

The power domain code shares the same node and will not probe if irqchip
probes first and marks the node with OF_POPULATED.

Clearing the OF_POPULATED flag is also done in imx_gpc_init for imx6 and
imx_gpcv2_irqchip_init implemented by upstream.

In imx_4.9.y this was solved in a different way by adding a second pgc
node, see commit fab513930e78 ("MLK-14280: gpc: gpc driver not probed").
Solving the problem by clearing OF_POPULATED allows using the upstream
PGC driver without hacks.

Having two irqchip implementations with same name seems to work fine
with the mach-imx variant taking precedence.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/gpcv2.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-imx/gpcv2.c b/arch/arm/mach-imx/gpcv2.c
index ef16e55..4174cb2 100644
--- a/arch/arm/mach-imx/gpcv2.c
+++ b/arch/arm/mach-imx/gpcv2.c
@@ -950,6 +950,12 @@ static int __init imx_gpcv2_init(struct device_node *node,
 	/* disable RBC */
 	imx_gpcv2_enable_rbc(false);
 
+	/*
+	 * Clear the OF_POPULATED flag set in of_irq_init so that
+	 * later the GPC power domain driver will not be skipped.
+	 */
+	of_node_clear_flag(node, OF_POPULATED);
+
 	return 0;
 }
 
-- 
1.7.9.5

